model/public/program.go

```diff
@@ -13,29 +16,51 @@ type Program struct {
 	Description         string
 	Cast                string
 	CategoryID          uint
-	Category            Category
-	Genres              []Genre `gorm:"many2many:genre_programs;"`
+	Category            Category `gorm:"foreignkey:CategoryID"`
+	Genres              []Genre  `gorm:"many2many:genre_programs;"`
 	SeasonID            uint
-	Season              Season
+	Season              Season `gorm:"foreignkey:SeasonID"`
 	SeasonEpisodeNumber int
 	LinearEpisodeNumber int
 	ProductionID        uint
-	Production          Production
+	Production          Production `gorm:"foreignkey:ProductionID"`
 	SerieID             uint
-	Serie               Serie
-	AirtimeFrom         int `gorm:"type:bigint"`
-	AirtimeTo           int `gorm:"type:bigint"`
-	Credit              []Credits
+	Serie               Serie     `gorm:"foreignkey:SerieID"`
+	AirtimeFrom         int       `gorm:"type:bigint"`
+	AirtimeTo           int       `gorm:"type:bigint"`
+	Credit              []Credits `gorm:"many2many:credit_groups;"`
+
 }
 
+//Find single program entry
 func (p Program) Find(id uint) Program {
-	db.Model(p).Where("id = ?", id).First(&p)
+	//Preload preloads structs - Creates a SQL query pr. Preload. Should be fixed in Gorm V2.
+	if err := db.Model(p).
+		Preload("Production").
+		Preload("Category").
+		Preload("Genres").
+		Preload("Serie").
+		Preload("Season").
+		Preload("Credit").
+		Where("id = ?", id).
+		First(&p).Error; err != nil {
+		fmt.Println(err)
```

Comment:
I stedet for bare at printe skal vi lige tænker over hvordan vi skal håndterer denne error?